src/server/dflycmd.cc

```diff
@@ -503,6 +525,15 @@ shared_ptr<DflyCmd::ReplicaInfo> DflyCmd::GetReplicaInfo(uint32_t sync_id) {
   return {};
 }
 
+std::vector<DflyCmd::ReplicaRoleInfo> DflyCmd::GetReplicasData() {
+  std::vector<ReplicaRoleInfo> vec;
+  unique_lock lk(mu_);
+  for (const auto& info : replica_infos_) {
+    vec.emplace_back(info.second->address, info.second->state);
```

Comment:
The ReplicaInfo fields are not guarded by the global mutex itself. While address is nowhere accessed, the state is on all operations. 

Reading from the state (and even reading an old value) has no consequences at all here, but doing this is technically undefined behavior, so we should probably make this field atomic